### ISSUE TITLE
Reserves counts as additional generation

### ISSUE THREAD

**Issue by ThanosGkou** [AntaresSimulatorTeam/Antares_Simulator#995]
I am using Antares 8.2.2 on Windows 10.
The issue is that the solution seems to report the reserves (day-ahead in my case) as additional generation.
Thus, the sum of all generated MWh, instead of being equal to the load, it is larger.

Simple case: a dummy day-ahead reserves timeseries = always 10% of load, will result in the generation being almost exactly 10% larger than load , on an annual basis at least. 

(Interconnections balance considered. The effect remains also in island mode cases)

**Comment by flomnes** [comment]
Hi thank you for submitting this issue. The Antares optimization is done in 3 steps

1. The day-ahead reserves are added to the load. For each thermal cluster, the optimizer decides the number of units that should be started at any hour. The adequacy problem is roughly `Production - Spillage + Unsupplied = Load + Day-ahead reserve`
2. The number of started units is rounded up using a specific heuristic and fixed
3. The optimizer decides the level of power for each thermal cluster between `Pmin` and `Pmax`. The adequacy problem is now `Production - Spillage + Unsupplied = Load`

In other words, the day-ahead reserve forces some thermal clusters to start even if it is not required to satisfy the demand. It may result in some spillage where and when the day-ahead reserve is high.

Is that clear for you ? Did you expect a different behavior ?

**Comment by ThanosGkou** [comment]
Hi,
Thanks for the comprehensive answer first of all.

As you mentioned, I would expect to see some spilled energy as well in order for this equality to hold, which is often not the case.
To speak with an example, I am running cases with "normal" and "high" RES, both of which, for the sake of simplicity, have day-ahead reserves = 10% of load. 
So, in the "normal" cases, there is no spilled/unserved energy, but still: Production = Load + 10%.
Also, in the "high-RES" scenarios, in cases where there is some spilled energy, still: production = Load + 10% + spilled 
, instead of: production = Load + spilled (ignoring unserved since it's zero in all cases)

So this seemed to me as a non-intentional behavior, where in some stage of the whole optimization process, the reserves are not re-subtracted from the load.

**Comment by PPlessiez** [comment]
Thank you for your answer, indeed your problem seems pretty clear, and as far as I know this is not a known issue.
Would you be able to attach an example study in which the problem occurs, so we can analyse it in more details?

**Comment by ThanosGkou** [comment]
I am having some difficulty reproducing this effect in a simple study. I'll investigate it a bit and come back.

**Comment by ThanosGkou** [comment]
I wasn't able to reproduce the issue starting from a blank study, so instead, I made an abstract version of my study which reproduces the issue. 
[democase.zip](https://github.com/AntaresSimulatorTeam/Antares_Simulator/files/10177906/democase.zip)

The setup is a single area with one load series, one hydro RoR+reservoir (no management),  two thermal super-clusters of 100 units each, and one cluster per pv, csp, wind.

Two solutions included, one with 0 reserves, and one with day-ahead reserves = 10% of load. 
In the "user" directory there are two excel files summing up the results on an annual level, where it shows that despite no spilled energy events, the generation is 10% greater than the load in the "withReserves" case. Of course you can verify that for yourselves.

**Comment by flomnes** [comment]
OK, I took a look at the study and your results.

It seems that when day-ahead reserves > 0, the thermal (gas) production increases to satisfy this requirement. This behavior is expected.

What I fail to understand is : where does this additional production go ? Since there is no spillage, it seems to be absorbed by the reserve. It's as if the reserve was added to the load. This behavior is not expected IMO.

**Comment by ThanosGkou** [comment]
Indeed it is seems there's something wrong. I think there might be something about the study configuration settings that makes it behave this way. I tried it in a blank study and it behaves properly.

**Comment by ThanosGkou** [comment]
Ok, the zip contains two, apparently identical studies: each one has solely a load series, reserves series, and two gas clusters. Nothing else.
The "peculiar" originates from my study
The "from_blank" was created starting from a blank study, but has the exact setup of the "peculiar" ( I compared every single parameter...)

So, the "peculiar" study outputs are (as discussed before): **generation > load** (with spilled = 0)
whereas the "from_blank" study outputs are: **generation = load** 

I can't see why in one case it behaves differently, when they are identical to one another. I start to suspect quantum mechanics :) 


[comparison.zip](https://github.com/AntaresSimulatorTeam/Antares_Simulator/files/10194935/comparison.zip)

**Comment by ThanosGkou** [comment]
Ok, found the cause!

I have been using my original study also for expansion planning. In the generaldata.ini, the "mode" was "Expansion" instead of "Economy" (although in the GUI, the mode was appearing as "Economy"). I assume this happens automatically when you run the expansion solver. 

When I manually changed the "mode" in the generaldata.ini file to "Economy", the behaviour was fixed (Load = Generation).

So, this is has to do with how the expansion solver sees reserves. I can imagine why this could be an intentional decision for the expansion solver. But if it's not, feel free to mention this as an issue.

**Comment by flomnes** [comment]
I will close this issue. The conclusion is that mode "Expansion" should be used carefully.

**Comment by ThanosGkou** [comment]
Agreed.
Just a final note. On closer look on the expansion documentation, it is indeed mentioned that when an expansion study is run with the setting "expansion_accurate", the antares solver runs in "Expansion" mode. (So it automatically sets the generaldata.ini mode to "Expansion" instead of "Economy", but in the GUI it still shows as "Economy".)

Perhaps it would be beneficial, that the study-updater would reset the generaldata.ini mode to "Economy" after the expansion finishes.
